framework/src/org/checkerframework/common/value/Range.java

```diff
@@ -0,0 +1,417 @@
+     *       when creating an annotation from range, which would then be replaced with
+     *       {@code @UnknownVal}.
+     * </ul>
+     *
+     * @param right the range to intersect with this range
+     * @return a range resulting from the intersection of the specified range and this range
+     */
+    public Range intersect(Range right) {
+        long resultFrom = Math.max(from, right.from);
+        long resultTo = Math.min(to, right.to);
+        return new Range(resultFrom, resultTo);
+    }
+
+    /**
+     * Returns the smallest range that includes all possible values resulting from adding an
+     * arbitrary value in the specified range to an arbitrary value in this range. We call this We
+     * call this the addition of two ranges.
+     *
+     * @param right a range to be added to this range
+     * @return the range resulting from the addition of the specified range and this range
+     */
+    public Range plus(Range right) {
+        long resultFrom = from + right.from;
+        long resultTo = to + right.to;
+        return new Range(resultFrom, resultTo);
+    }
+
+    /**
+     * Returns the smallest range that includes all possible values resulting from subtracting an
+     * arbitrary value in the specified range from an arbitrary value in this range. We call this
+     * the subtraction of two ranges.
+     *
+     * @param right the range to be subtracted from this range
+     * @return the range resulting from subtracting the specified range from this range
+     */
+    public Range minus(Range right) {
+        long resultFrom = from - right.to;
+        long resultTo = to - right.from;
+        return new Range(resultFrom, resultTo);
+    }
+
+    /**
+     * Returns the smallest range that includes all possible values resulting from multiplying an
+     * arbitrary value in the specified range by an arbitrary value in this range. We call this the
+     * multiplication of two ranges.
+     *
+     * @param right the specified range to be multiplied by this range
+     * @return the range resulting from multiplying the specified range by this range
+     */
+    public Range times(Range right) {
+        ArrayList<Long> possibleValues =
+                Arrays.asList(from * right.from, from * right.to, to * right.from, to * right.to);
+        return new Range(Collections.min(possibleValues), Collections.max(possibleValues));
+    }
+
+    /**
+     * Returns the smallest range that includes all possible values resulting from dividing (integer
+     * division) an arbitrary value in this range by an arbitrary value in the specified range. We
+     * call this the division of two ranges.
+     *
+     * @param right the specified range by which this range is divided
+     * @return the range resulting from dividing this range by the specified range
+     */
+    public Range divide(Range right) {
+        long resultFrom = Long.MIN_VALUE;
```

Comment:
If the variable is guaranteed to be initialized later, then it's dead code (and more importantly is confusing to a reader!) to initialize it in the declaration.  Please remove the initializers for resultFrom and resultTo.